### PR TITLE
Data preprocessing page

### PR DESCRIPTION
**Pull request by thanhnew2001**
- Allow to change data type and title
- Data type is still in text as no combo box in TaipyGUI
- Re-generate suggested prompts when column title changes

### CODE REVIEW

File: utils.py [hunk 1]
```diff
@@ -97,6 +97,27 @@ def beautify_labels(df):
     df.columns = new_columns
 
 
+def generate_completion(prompt):
```
Comment (AlexandreSajus) [inline utils.py:100]: Why is this function needed?

File: utils.py [hunk 2]
```diff
@@ -136,66 +157,95 @@ def categorize_columns_by_datatype(data_frame):
...
+    "Render",
+    "Depict",
+    "Produce",
+    "Craft",
+    "Chart",
+    "Illustrate",
+]
 
 
-def generate_prompts(string_columns, float_columns, date_columns, int_columns):
```
Comment (AlexandreSajus) [inline utils.py:179]: Why was my approach to generating prompts scrapped (the one in my last PR)?

**Review by AlexandreSajus — CHANGES_REQUESTED** [review]
Good iteration! The new page is almost what we need.

Some things we will need to fix (I will keep this PR open so we can work on it):
- Types should be chosen from a dropdown menu instead of a text input
- When a conversion to a type fails, we should give the choice to the user to cancel or remove rows that are problematic
- We probably need a UI / Layout pass
- Why were my changes to prompt generation scrapped in this PR?

File: utils.py (hunk 2 above)
Comment (thanhnew2001) [reply utils.py]: I moved this into util.py

File: utils.py (hunk 1 above)
Comment (thanhnew2001) [reply utils.py]: This can be removed. Initially, I planned to use openAI to suggest good prompts but they request a paid account

**Comment by thanhnew2001** [comment]
Types should be chosen from a dropdown menu instead of a text input 
--> Currently, TaipyGUI does not support it. Haven't found a better way
When a conversion to a type fails, we should give the choice to the user to cancel or remove rows that are problematic
--> Sure, will take some time to rework that way
We probably need a UI / Layout pass
--> Yes, definitely, TaipyGUI has some limits but we can find a workaround
Why were my changes to prompt generation scrapped in this PR?
--> Sorry, I moved some functions around. Let me know if anything is really missing.

**Comment by AlexandreSajus** [comment]
Two missing items left:
- Skip problematic rows when type conversion fails, ask/notify the user
- UX pass using an editable table:

We should replace this layout:
image

With an editable table that contains two rows: columns names and column type
on_edit we will change the column names and types

**Comment by AlexandreSajus** [comment]
Changing column names and types is now done through an editable table, making the UX more appealing.

Two missing items:
- If a type conversion fails, the user should be given the choice to ignore rows where the conversion fails
- Some buttons (Reset App, Reset Columns) need fixing since we have added new variables to the app

**Comment by AlexandreSajus** [comment]
The branch should be stable now. Fixed the following issues:
- :wrench: Reset data before a new prompt execution to avoid prompting on a dataset missing columns
- :wrench: Rewrote the initial prompt suggestions
- :bug: Reset Data button should not reset the data preprocessing
- :bug: Reset App fails
- :bug: Reset App after uploading a dataset shows 10 prompt suggestions
- :wrench: Switch type names to lowercase
- :bug: Prompt builder should reset after editing column names or types
- :bug: Prompt suggestions should reset after editing column names or types
- :memo: Add a message to ask the user to refresh the page if generating a graph fails

**Review by AlexandreSajus — APPROVED** [review]

**Comment by AlexandreSajus** [comment]
Fixes #55 
Fixes #60 
Fixes #63

**Pull request merged**